central/debug/service/service.go

```diff
@@ -625,6 +630,11 @@ func (s *serviceImpl) getDebugDump(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if env.DisableCentralDiagnostics.BooleanSetting() {
+		// Regardless of query, don't generate logs for managed central
+		opts.logs = noLogs
```

Comment:
You would also need to set `opts.withLogImbue` to `false` per what is said in the ticket, but see my comments above - I think it would be better to push that check down one level to where we actually attempt to collect logs.